src/WorkerChannel.ts

```diff
@@ -46,11 +47,21 @@ export class WorkerChannel implements IWorkerChannel {
       if (eventHandler) {
         eventHandler.apply(this, [msg.requestId, msg[event]]);
       } else {
-        systemError(`Worker ${workerId} had no handler for message '${event}'`)
+        this.log({
+          message: `Worker ${workerId} had no handler for message '${event}'`,
+          level: LogLevel.Error,
+          logCategory: LogCategory.System
+        });
       }
     });
     eventStream.on('error', function (err) {
-      systemError(`Worker ${workerId} encountered event stream error: `, err);
+      eventStream.write({

```

Comment:
I think this should be a system log as well because the event stream itself is in error. it's unlikely that the event stream recovered